### DISCUSSION TITLE
Regarding tap-s3 connector

### DISCUSSION THREAD

**Discussion by silverbullet1** [meltano/meltano#8318] [Q&A]
Hi all,
I wanted to extract data from my s3 source but the catch is that data is in parquet format. I wanted to know what is the best way to extract it ? We only have tap-s3-csv connector available from official hub.
I could think of 2 approaches:
1. Create a new tap tap-s3-parquet and take care of decoding parquet data to JSON
2. Use some transformer(?) or another tool in the first step which converts parquet to csv and then use tap-s3-csv connector without any changes.

my target is postgres

Please help me out :) Thanks

**Comment by edgarrmondragon** [comment]
It may be a relatively low effort to support Parquet in https://github.com/MeltanoLabs/tap-universal-file

**Reply by tayloramurphy** [reply]
@visch how hard would it be to add that? Indeed - we already have an issue for that https://github.com/MeltanoLabs/tap-universal-file/issues/41 😄

**Reply by visch** [reply]
Pretty low effort! Half a day maybe?
